src/contrib/persistence/Akka.Persistence.SqlServer/Journal/QueryBuilder.cs

```diff
@@ -0,0 +1,161 @@
+﻿using System.Data;
+using System.Data.SqlClient;
+using System.Text;
+
+namespace Akka.Persistence.SqlServer.Journal
+{
+    /// <summary>
+    /// SQL query builder used for generating queries required to perform journal's tasks.
+    /// </summary>
+    public interface IJournalQueryBuilder
+    {
+        /// <summary>
+        /// Returns query which should return a frame of messages filtered accordingly to provided parameters.
+        /// </summary>
+        SqlCommand SelectMessages(string persistenceId, long fromSequenceNr, long toSequenceNr, long max);
+
+        /// <summary>
+        /// Returns query returning single number considered as the highest sequence number in current journal.
+        /// </summary>
+        SqlCommand SelectHighestSequenceNr(string persistenceId);
+
+        /// <summary>
+        /// Returns a non-query command used to insert collection of <paramref name="messages"/> in journal table.
+        /// </summary>
+        SqlCommand InsertBatchMessages(IPersistentRepresentation[] messages);
+
+        /// <summary>
+        /// Depending on <paramref name="permanent"/> flag this method may return either UPDATE or DELETE statement
+        /// used to alter IsDeleted field or delete rows permanently.
+        /// </summary>
+        SqlCommand DeleteBatchMessages(string persistenceId, long toSequenceNr, bool permanent);
+    }
+
+    internal class DefaultJournalQueryBuilder : IJournalQueryBuilder
+    {
+        private readonly string _schemaName;
+        private readonly string _tableName;
+
+        private readonly string _selectHighestSequenceNrSql;
+        private readonly string _insertMessagesSql;
+
+        public DefaultJournalQueryBuilder(string tableName, string schemaName)
+        {
+            _tableName = tableName;
+            _schemaName = schemaName;
+
+            _insertMessagesSql =
+                string.Format("INSERT INTO [{0}].[{1}] (PersistenceID, SequenceNr, IsDeleted, PayloadType, Payload) VALUES (@PersistenceId, @SequenceNr, @IsDeleted, @PayloadType, @Payload)",
+                    _schemaName, _tableName);
+            _selectHighestSequenceNrSql = string.Format(@"SELECT MAX(SequenceNr) FROM [{0}].[{1}] WHERE CS_PID = CHECKSUM(@pid)", _schemaName, _tableName);
```

Comment:
Maybe a non-issue but this is open for sql injection incase someone manages to manipulate tableNAme or schemaName.
(if that is the case the user is already in trouble, but maybe we should throw some validation at this just to be sure)